### PR TITLE
Watching plus Unicode

### PR DESCRIPTION
**Pull request by jglick**
Integrates #60 & #61. Please leave line comments in one of the parents until you are referring specifically to something about the integration.

@reviewbybees

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@jglick I have approved #61. #60 is WiP if I understand your comment correctly. I'd guess no need to review this PR now.

**Comment by jglick** [comment]
Please do _not_ perform line-by-line reviews in this PR. Rather, review component PRs.

File: src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java [hunk 1]
```diff
@@ -272,7 +369,111 @@ public FilePath getOutputFile(FilePath workspace) throws IOException, Interrupte
...
+                    lastLocation = Long.parseLong(lastLocationFile.readToString());
+                }
+                FilePath logFile = controller.getLogFile(workspace);
+                long len = logFile.length();
+                if (len > lastLocation) {
+                    assert !logFile.isRemote();
+                    try (FileChannel ch = FileChannel.open(Paths.get(logFile.getRemote()), StandardOpenOption.READ)) {
+                        InputStream locallyEncodedStream = Channels.newInputStream(ch.position(lastLocation));
+                        InputStream utf8EncodedStream = cs == null ? locallyEncodedStream : new ReaderInputStream(new InputStreamReader(locallyEncodedStream, cs), StandardCharsets.UTF_8);
+                        CountingInputStream cis = new CountingInputStream(utf8EncodedStream);
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java]: This probably does the same miscount when transcoding as #61, to be confirmed.

**Comment by jglick** [comment]
Closing as this is now identical to #60.

**Pull request closed without merging**